src/com/facebook/buck/cli/TestCommand.java

```diff
@@ -691,4 +712,60 @@ public void printUsage(PrintStream stream) {
   public String getShortDescription() {
     return "builds and runs the tests for the specified target";
   }
+
+  /**
+   * args4j does not support parsing repeated (or delimiter separated) Enums by default.
+   * {@link CoverageReportFormatsHandler} implements args4j behavior for CoverageReportFormat.
+   */
+  public static class CoverageReportFormatsHandler extends OptionHandler<CoverageReportFormat> {
+
+    public CoverageReportFormatsHandler(
+        CmdLineParser parser, OptionDef option, Setter<CoverageReportFormat> setter) {
+      super(parser, option, setter);
+    }
+
+    @Override
+    public int parseArguments(Parameters params) throws CmdLineException {
+      Set<String> parsed =
+          Splitter.on(",")
+              .splitToList(params.getParameter(0))
+              .stream()
+              .map(s -> s.replaceAll("-", "_").toLowerCase())
+              .collect(Collectors.toSet());
+      List<CoverageReportFormat> formats = new ArrayList<>();
+      for (CoverageReportFormat format : CoverageReportFormat.values()) {
+        if (parsed.remove(format.name().toLowerCase())) {
+          formats.add(format);
+        }
+      }
+
+      if (parsed.size() != 0) {
+        String invalidFormats = parsed.stream().collect(Collectors.joining(","));
+        if (option.isArgument()) {
+          throw new CmdLineException(
+              owner, Messages.ILLEGAL_OPERAND, option.toString(), invalidFormats);
+        } else {
+          throw new CmdLineException(
+              owner, Messages.ILLEGAL_OPERAND, params.getParameter(-1), invalidFormats);
+        }
+      }
+
+      for (CoverageReportFormat format : formats) {
+        setter.addValue(format);
+      }
+      return 1;
```

Comment:
this was formats.size() in one of the previous revisions, which is incorrect. For csv arg we always consume only single one.